[2024-05-13 15:22:19,823: INFO: main: Welcome to our custom logging]
[2024-05-17 14:56:33,671: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-17 14:56:33,676: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-17 14:59:27,287: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-17 14:59:27,301: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-17 14:59:54,035: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-17 14:59:54,043: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-17 14:59:54,046: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-17 14:59:54,049: INFO: common: created directory at: artifacts]
[2024-05-17 14:59:54,053: INFO: common: created directory at: artifacts/data_ingestion]
[2024-05-17 14:59:55,032: INFO: 2366278714: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 23FE:0880:175CEE:1EF255:66472391
Accept-Ranges: bytes
Date: Fri, 17 May 2024 09:29:54 GMT
Via: 1.1 varnish
X-Served-By: cache-hyd1100025-HYD
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1715938194.378182,VS0,VE288
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: e506c52431429d54ebb5797d7943ad49cbf1b75c
Expires: Fri, 17 May 2024 09:34:54 GMT
Source-Age: 0

]
[2024-05-21 10:18:06,948: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-05-21 10:18:06,971: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 10:18:06,983: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 10:18:06,986: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 10:18:06,987: INFO: common: created directory at: artifacts]
[2024-05-21 10:18:06,988: INFO: common: created directory at: artifacts/data_ingestion]
[2024-05-21 10:18:07,097: ERROR: main: <urlopen error [Errno 11001] getaddrinfo failed>]
Traceback (most recent call last):
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 1344, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1319, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1365, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1314, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1074, in _send_output
    self.send(msg)
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1018, in send
    self.connect()
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 1453, in connect
    super().connect()
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\http\client.py", line 984, in connect
    self.sock = self._create_connection(
                ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\socket.py", line 828, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\socket.py", line 963, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno 11001] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\Gate\DS\End to End ML\End-to-End-Machine-Learning-Project-with-MLFlow\main.py", line 8, in <module>
    data_ingestion.main()
  File "D:\Gate\DS\End to End ML\End-to-End-Machine-Learning-Project-with-MLFlow\src\mlProject\pipeline\stage_01_data_ingestion.py", line 17, in main
    data_ingestion.download_file()
  File "D:\Gate\DS\End to End ML\End-to-End-Machine-Learning-Project-with-MLFlow\src\mlProject\components\data_ingestion.py", line 18, in download_file
    filename, headers = request.urlretrieve(
                        ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 240, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
                            ^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 215, in urlopen
    return opener.open(url, data, timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 515, in open
    response = self._open(req, data)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 532, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 492, in _call_chain
    result = func(*args)
             ^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 1392, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\DELL\AppData\Local\Programs\Python\Python312\Lib\urllib\request.py", line 1347, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno 11001] getaddrinfo failed>
[2024-05-21 10:20:11,819: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-05-21 10:20:11,825: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 10:20:11,827: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 10:20:11,828: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 10:20:11,829: INFO: common: created directory at: artifacts]
[2024-05-21 10:20:11,830: INFO: common: created directory at: artifacts/data_ingestion]
[2024-05-21 10:20:13,248: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: B8D8:41EF8:53A7CF:6432FE:664C2803
Accept-Ranges: bytes
Date: Tue, 21 May 2024 04:50:13 GMT
Via: 1.1 varnish
X-Served-By: cache-bom4742-BOM
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1716267013.065407,VS0,VE420
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 337fe75eccdfe57ae006454cf31f130579895df0
Expires: Tue, 21 May 2024 04:55:13 GMT
Source-Age: 0

]
[2024-05-21 10:20:13,271: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-05-21 11:10:30,486: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-05-21 11:10:30,491: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 11:10:30,494: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 11:10:30,498: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 11:10:30,499: INFO: common: created directory at: artifacts]
[2024-05-21 11:10:30,500: INFO: common: created directory at: artifacts/data_ingestion]
[2024-05-21 11:10:31,934: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: B8D8:41EF8:53A7CF:6432FE:664C2803
Accept-Ranges: bytes
Date: Tue, 21 May 2024 05:40:32 GMT
Via: 1.1 varnish
X-Served-By: cache-bom4737-BOM
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1716270032.692872,VS0,VE330
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: abe2f6f8a62aeca91a9505a3e8e0f168e26a4755
Expires: Tue, 21 May 2024 05:45:32 GMT
Source-Age: 0

]
[2024-05-21 11:10:31,956: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-05-21 11:11:35,741: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 11:11:35,746: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 11:11:35,753: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 11:11:35,756: INFO: common: created directory at: artifacts]
[2024-05-21 11:12:44,918: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 11:12:44,921: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 11:12:44,930: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 11:12:44,934: INFO: common: created directory at: artifacts]
[2024-05-21 11:12:44,937: INFO: common: created directory at: artifacts/data_validation]
[2024-05-21 11:28:18,002: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 11:28:18,006: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 11:28:18,012: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 11:28:18,016: INFO: common: created directory at: artifacts]
[2024-05-21 11:28:18,019: INFO: common: created directory at: artifacts/data_validation]
[2024-05-21 12:05:14,116: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-05-21 12:05:14,121: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 12:05:14,123: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 12:05:14,125: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 12:05:14,127: INFO: common: created directory at: artifacts]
[2024-05-21 12:05:14,128: INFO: common: created directory at: artifacts/data_ingestion]
[2024-05-21 12:05:16,119: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: B8D8:41EF8:53A7CF:6432FE:664C2803
Accept-Ranges: bytes
Date: Tue, 21 May 2024 06:35:16 GMT
Via: 1.1 varnish
X-Served-By: cache-bom4734-BOM
X-Cache: HIT
X-Cache-Hits: 0
X-Timer: S1716273316.663727,VS0,VE461
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 31c1253289004f2fc654efe1a17dbf6ce6c700bb
Expires: Tue, 21 May 2024 06:40:16 GMT
Source-Age: 0

]
[2024-05-21 12:05:16,145: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-05-21 12:05:16,145: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2024-05-21 12:05:16,146: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-21 12:05:16,150: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-21 12:05:16,150: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-21 12:05:16,150: INFO: common: created directory at: artifacts]
[2024-05-21 12:05:16,150: INFO: common: created directory at: artifacts/data_validation]
[2024-05-21 12:05:16,182: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2024-05-22 13:58:35,598: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 13:58:35,613: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 13:58:35,621: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 13:58:35,625: INFO: common: created directory at: artifacts]
[2024-05-22 13:58:35,627: INFO: common: created directory at: artifacts/data_transformation]
[2024-05-22 14:05:47,363: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:05:47,367: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:05:47,371: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:05:47,374: INFO: common: created directory at: artifacts]
[2024-05-22 14:05:47,378: INFO: common: created directory at: artifacts/data_transformation]
[2024-05-22 14:05:47,466: INFO: 3503125014: Splited data into training and test sets]
[2024-05-22 14:05:47,467: INFO: 3503125014: (1199, 12)]
[2024-05-22 14:05:47,467: INFO: 3503125014: (400, 12)]
[2024-05-22 14:14:53,443: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-05-22 14:14:53,455: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:14:53,471: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:14:53,473: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:14:53,483: INFO: common: created directory at: artifacts]
[2024-05-22 14:14:53,483: INFO: common: created directory at: artifacts/data_ingestion]
[2024-05-22 14:14:53,483: INFO: data_ingestion: File already exists of size: ~ 23 KB]
[2024-05-22 14:14:53,508: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-05-22 14:14:53,508: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2024-05-22 14:14:53,512: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:14:53,512: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:14:53,516: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:14:53,516: INFO: common: created directory at: artifacts]
[2024-05-22 14:14:53,516: INFO: common: created directory at: artifacts/data_validation]
[2024-05-22 14:14:53,551: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2024-05-22 14:14:53,551: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-05-22 14:14:53,561: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:14:53,561: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:14:53,566: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:14:53,566: INFO: common: created directory at: artifacts]
[2024-05-22 14:14:53,566: INFO: common: created directory at: artifacts/data_transformation]
[2024-05-22 14:14:53,602: INFO: data_transformation: Splited data into training and test sets]
[2024-05-22 14:14:53,602: INFO: data_transformation: (1199, 12)]
[2024-05-22 14:14:53,602: INFO: data_transformation: (400, 12)]
[2024-05-22 14:14:53,602: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-05-22 14:21:19,930: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:21:19,956: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:21:19,963: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:21:19,969: INFO: common: created directory at: artifacts]
[2024-05-22 14:21:19,972: INFO: common: created directory at: artifacts/model_trainer]
[2024-05-22 14:21:33,134: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:21:33,137: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:21:33,142: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:21:33,145: INFO: common: created directory at: artifacts]
[2024-05-22 14:21:33,147: INFO: common: created directory at: artifacts/model_trainer]
[2024-05-22 14:29:34,008: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-05-22 14:29:34,023: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:29:34,023: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:29:34,023: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:29:34,034: INFO: common: created directory at: artifacts]
[2024-05-22 14:29:34,034: INFO: common: created directory at: artifacts/data_ingestion]
[2024-05-22 14:29:36,317: INFO: data_ingestion: artifacts/data_ingestion/data.zip download! with following info: 
Connection: close
Content-Length: 23329
Cache-Control: max-age=300
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Content-Type: application/zip
ETag: "c69888a4ae59bc5a893392785a938ccd4937981c06ba8a9d6a21aa52b4ab5b6e"
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
X-GitHub-Request-Id: 548E:56EB6:4F9A48:69F3C6:664DB3F8
Accept-Ranges: bytes
Date: Wed, 22 May 2024 08:59:36 GMT
Via: 1.1 varnish
X-Served-By: cache-bom4727-BOM
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1716368376.394466,VS0,VE268
Vary: Authorization,Accept-Encoding,Origin
Access-Control-Allow-Origin: *
Cross-Origin-Resource-Policy: cross-origin
X-Fastly-Request-ID: 9ca156fcdf364b61a24d44726d023ae4fc1699c3
Expires: Wed, 22 May 2024 09:04:36 GMT
Source-Age: 0

]
[2024-05-22 14:29:36,344: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-05-22 14:29:36,344: INFO: main: >>>>>> stage Data Validation stage started <<<<<<]
[2024-05-22 14:29:36,348: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:29:36,351: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:29:36,352: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:29:36,354: INFO: common: created directory at: artifacts]
[2024-05-22 14:29:36,354: INFO: common: created directory at: artifacts/data_validation]
[2024-05-22 14:29:36,385: INFO: main: >>>>>> stage Data Validation stage completed <<<<<<

x==========x]
[2024-05-22 14:29:36,385: INFO: main: >>>>>> stage Data Transformation stage started <<<<<<]
[2024-05-22 14:29:36,401: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:29:36,403: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:29:36,403: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:29:36,406: INFO: common: created directory at: artifacts]
[2024-05-22 14:29:36,406: INFO: common: created directory at: artifacts/data_transformation]
[2024-05-22 14:29:36,464: INFO: data_transformation: Splited data into training and test sets]
[2024-05-22 14:29:36,479: INFO: data_transformation: (1199, 12)]
[2024-05-22 14:29:36,479: INFO: data_transformation: (400, 12)]
[2024-05-22 14:29:36,479: INFO: main: >>>>>> stage Data Transformation stage completed <<<<<<

x==========x]
[2024-05-22 14:29:36,479: INFO: main: >>>>>> stage Model Trainer stage started <<<<<<]
[2024-05-22 14:29:36,479: INFO: common: yaml file: config\config.yaml loaded successfully]
[2024-05-22 14:29:36,479: INFO: common: yaml file: params.yaml loaded successfully]
[2024-05-22 14:29:36,490: INFO: common: yaml file: schema.yaml loaded successfully]
[2024-05-22 14:29:36,495: INFO: common: created directory at: artifacts]
[2024-05-22 14:29:36,495: INFO: common: created directory at: artifacts/model_trainer]
[2024-05-22 14:29:36,543: INFO: main: >>>>>> stage Model Trainer stage completed <<<<<<

x==========x]
